### PR TITLE
add support for snakemake directory inputs

### PR DESCRIPTION
**Pull request by rahuldesai1**

### CODE REVIEW

File: latch_cli/snakemake/serialize_utils.py [hunk 1]
```diff
@@ -217,9 +217,7 @@ def get_serializable_workflow(
...
-    if cur.is_file():
-        mapping[str(stem)] = remote
-        return
-
-    for p in cur.iterdir():
-        update_mapping(p, stem / p.name, urljoins(remote, p.name), mapping)
+    if cur.is_dir():
+        for p in cur.iterdir():
+            update_mapping(p, stem / p.name, urljoins(remote, p.name), mapping)
+    mapping[str(stem)] = remote
```
Comment (AidanAbd) [inline latch_cli/snakemake/serialize_utils.py:223]: Not file != Is Dir, that is the reason for this?

File: latch_cli/snakemake/workflow.py [hunk 2]
```diff
@@ -84,6 +85,10 @@
 SnakemakeInputVal: TypeAlias = snakemake.io._IOFile
 
 
+def _is_dir(x: SnakemakeInputVal) -> bool:
+    return os.path.isdir(x)
```
Comment (AidanAbd) [inline latch_cli/snakemake/workflow.py:89]: kinda odd to have a one line function here instead of using `os.path.isdir` in place of the function calls.

**Review by AidanAbd — APPROVED** [review]

File: latch_cli/snakemake/workflow.py (hunk 2 above)
Comment (rahuldesai1) [reply latch_cli/snakemake/workflow.py:89]: True, I initially had it as 
`return os.path.isdir(x) or x.is_directory`
but realized that idk what x.is_directory even means ( I think its some Snakemake specific construct that seems to return false when x is a directory so remove that check. 

will inline this

Comment (AidanAbd) [reply latch_cli/snakemake/workflow.py:89]: sgsg

File: latch_cli/snakemake/serialize_utils.py (hunk 1 above)
Comment (rahuldesai1) [reply latch_cli/snakemake/serialize_utils.py:223]: Yeah, it only really makes sense to `iterdir` on directories. 

for example, if `cur` points to a device file, we shouldn't iterdir over that file. Symlinks should resolve to the underlying file so it should work as expected for those as well.

**Review by ayushkamat — APPROVED** [review]

**Pull request merged**